MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift

```diff
@@ -150,3 +150,54 @@ public enum MASDKCustomPermissionError: String, MiniAppErrorProtocol {
         }
     }
 }
+
+/// Enumeration that is used to return  Access Token error
+public enum MASDKAccessTokenError: Error, MiniAppErrorProtocol {
+
+    /// Host app failed to implement required interface
+    case failedToConformToProtocol
+
+    /// Requested Audience is not supported
+    case audienceNotSupportedError
+
+    /// Requested Scope is not supported
+    case scopesNotSupportedError
+
+    /// Authorization failed and the reason will be shared by the host app
+    case authorizationFailureError(description: String)
+
+    /// Unknown/Custom error
+    case error(description: String)
+
+    /// Detailed Description for every MASDKAccessTokenError
+    public var description: String {
+        switch self {
+        case .failedToConformToProtocol:
+            return MASDKLocale.localize(.failedToConformToProtocol)
+        case .audienceNotSupportedError:
+            return "The value passed for 'audience' is not supported."
+        case .scopesNotSupportedError:
+            return "The value passed for 'scopes' is not supported."
+        case .authorizationFailureError(let description):
+            return description
+        case .error(let description):
```

Comment:
in JS SDK we called this kind of errors "Other", do you think we can bring this to our own SDK?

Reply:
Yeah, not a problem. But I sort of followed the confluence documentation. Should we update the documentation as well ?

Reply:
I think that's not that important, as what I did on JS side is that if I receive an unknown error I set is as `Other` type. We can go with this implementation.